src/chart_types/xy_chart/utils/axis_utils.ts

```diff
@@ -400,7 +400,7 @@ export function getAvailableTicks(
   totalBarsInCluster: number,
   enableHistogramMode: boolean,
 ): AxisTick[] {
-  const ticks = scale.ticks();
+  const ticks = axisSpec.integersOnly ? scale.ticks().filter((item) => item % 1 === 0) : scale.ticks();
```

Comment:
since `ticks()` returns type `any[]` you may want to add `typeof item === 'number'` to the check
```suggestion
  const ticks = axisSpec.integersOnly ? scale.ticks().filter((item) => typeof item === 'number' && item % 1 === 0) : scale.ticks();
```

Reply:
This makes so much sense! I'm adding this in commit 000a4ac